File: backend/internal/models/journal.go
package models

import (
	"time"
	"github.com/google/uuid"
	"gorm.io/gorm"
)

type JournalEntry struct {
	ID          uuid.UUID      `gorm:"type:uuid;default:gen_random_uuid();primaryKey" json:"id"`
	UserID      uuid.UUID      `gorm:"type:uuid;index" json:"user_id"`
	MoodEmoji   string         `gorm:"type:varchar(10)" json:"mood_emoji"`
	MoodScore   int            `gorm:"default:50" json:"mood_score"`
	Content     string         `gorm:"type:text" json:"content"`
	PhotoURL    string         `gorm:"type:text" json:"photo_url"`
	CardColor   string         `gorm:"type:varchar(7)" json:"card_color"`
	EntryDate   time.Time      `gorm:"index" json:"entry_date"`
	IsPrivate   bool           `gorm:"default:true" json:"is_private"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`
}

type JournalStreak struct {
	ID            uuid.UUID `gorm:"type:uuid;default:gen_random_uuid();primaryKey" json:"id"`
	UserID        uuid.UUID `gorm:"type:uuid;uniqueIndex" json:"user_id"`
	CurrentStreak int       `gorm:"default:0" json:"current_streak"`
	LongestStreak int       `gorm:"default:0" json:"longest_streak"`
	TotalEntries  int       `gorm:"default:0" json:"total_entries"`
	LastEntryDate time.Time `json:"last_entry_date"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}

var MoodEmojis = []string{"😊", "😢", "😡", "😰", "😴", "🥳", "😌", "🤔", "😍", "😤"}
var CardColors = []string{"#fef3c7", "#dbeafe", "#dcfce7", "#fce7f3", "#ede9fe", "#fef2f2"}

File: mobile/types/journal.ts
export interface JournalEntry {
  id: string;
  user_id: string;
  mood_emoji: string;
  mood_score: number;
  content: string;
  photo_url: string;
  card_color: string;
  entry_date: string;
  is_private: boolean;
  created_at: string;
  updated_at: string;
}

export interface JournalStreak {
  id: string;
  user_id: string;
  current_streak: number;
  longest_streak: number;
  total_entries: number;
  last_entry_date: string;
}

export interface JournalListResponse {
  entries: JournalEntry[];
  total: number;
  limit: number;
  offset: number;
}

export interface WeeklyInsights {
  average_mood_score: number;
  mood_trend: string;
  top_mood: string;
  total_entries: number;
  daily_scores: DailyScore[];
}

export interface DailyScore {
  date: string;
  score: number;
}

export interface GuestEntry {
  id: string;
  mood_emoji: string;
  mood_score: number;
  content: string;
  card_color: string;
  created_at: string;
}
